### DISCUSSION TITLE
How is Terminal able to use XAML Islands to host a TabView, which is in the Microsoft namespace?

### DISCUSSION THREAD

**Discussion by amtopel** [microsoft/terminal#14739] [Q&A]
Hi, this isn't an issue, just a question.  I've been reviewing the source code but I'm unable to understand it.  Can anyone explain how Terminal is able to use XAML Islands to host a Tabview, which is in the Microsoft namespace?  I was under the impression that XAML Islands can only host stuff from the Windows namespace.  But I'm also thoroughly confused by all these frameworks.  Can anyone shed any light on this?  Thanks for any info.

**Comment by zadjii-msft** [comment]
> I was under the impression that XAML Islands can only host stuff from the Windows namespace

Nope! But you will need the Microsoft.Ui.Xaml nuget package, and an `App` instance (to help load the MUX XAML & resources), and wiring that up can be... tricky. 

Fortunately, we have a scratch project that's (fairly) straightforward which does all that! Take a look at the projects in https://github.com/microsoft/terminal/tree/main/scratch/ScratchIslandApp. Those are all built from https://github.com/microsoft/terminal/blob/main/Scratch.sln. 

There's a couple helper project files scattered around the repo which that sln depends on, but that should be a good place to start!
